src/components/ui/Panel.jsx

```diff
@@ -66,6 +66,9 @@ export default class Panel extends React.Component {
 
   componentDidMount() {
     this.updateMobileMapUI();
+    this.setState({
```

Comment:
It's generally not a good idea to modify the state synchronously in `componentDidMount` as it will trigger immediately a new render after the first one.
See also next comment.